### PR TITLE
ZBUG-1922 Shared folder Navigation issue in ModernUI

### PR DESCRIPTION
**Pull request by SohamBiswasTelus**
-Made changes in Zimbra-in-memory-cache.ts for creating unique id for caching.

-Made changes in batch-client/index.ts
In updateAbsoluteFolderPath

### CODE REVIEW

File: src/apollo/zimbra-in-memory-cache.ts [hunk 1]
```diff
@@ -22,10 +22,13 @@ const dataIdFromObject = (object: any) => {
...
 				return `${defaultDataIdFromObject(object)}:local`;
-			}
-			if (object.id === '1') {
+			} else if (object.id === '1') {
 				// Cache the root folder based on both ID and UUID from server
 				return `${object.__typename}:${object.id}:${object.uuid}`;
+			} else if (object?.id?.includes(':')) {
+				// Cache the root folder based on both ID and UUID from server
+				let i = `${object.id}:${object.absFolderPath}`;
+				return i;
```
Comment (silentsakky) [inline src/apollo/zimbra-in-memory-cache.ts:31]: - [ ] combine both statements
- [ ] also fix comment
- [ ] absFolderPath can contain special characters, which can break caching mechanism, how would you handle that?

File: src/batch-client/index.ts [hunk 2]
```diff
@@ -233,13 +233,12 @@ function updateAbsoluteFolderPath(originalName: any, parentFolderAbsPath: string
 		if (originalName === 'USER_ROOT') {
 			folder.absFolderPath = `${parentFolderAbsPath}${folder.absFolderPath}`;
 		} else {
-			folder.absFolderPath = folder.absFolderPath.replace(`/${originalName}`, parentFolderAbsPath);
+			folder.absFolderPath = `${parentFolderAbsPath}/${folder?.name}`;
```
Comment (silentsakky) [inline src/batch-client/index.ts:236]: do we need optional chaining operator here?

Comment (SohamBiswasTelus) [reply src/batch-client/index.ts]: I have changed the code.

File: src/apollo/zimbra-in-memory-cache.ts (hunk 1 above)
Comment (SohamBiswasTelus) [reply src/apollo/zimbra-in-memory-cache.ts]: First two comments code change is done.
For third point I have checked with `${{(PAE??SI**~1!@#%<><>))+^==^` seems like working fine. And we cannot use colon, slash and double-quote in name

File: src/apollo/zimbra-in-memory-cache.ts [hunk 3]
```diff
@@ -22,10 +22,12 @@ const dataIdFromObject = (object: any) => {
 		case 'Folder':
 			if (object.local && object.local !== null) {
 				return `${defaultDataIdFromObject(object)}:local`;
-			}
-			if (object.id === '1') {
+			} else if (object.id === '1') {
 				// Cache the root folder based on both ID and UUID from server
 				return `${object.__typename}:${object.id}:${object.uuid}`;
+			} else if (object?.id?.includes(':')) {
```
Comment (silentsakky) [inline src/apollo/zimbra-in-memory-cache.ts:28]: not sure if we need optional chaining operator here

**Review by silentsakky — APPROVED** [review]

**Pull request merged**